lib/wasix/src/runners/wasi.rs

```diff
@@ -367,22 +433,35 @@ impl crate::runners::Runner for WasiRunner {
         };
 
         #[allow(unused_mut)]
-        let mut env = self
+        let mut builder = self
             .prepare_webc_env(exec_name, &wasi, Some(pkg), Arc::clone(&runtime), None)
             .context("Unable to prepare the WASI environment")?;
 
         #[cfg(feature = "journal")]
         {
             for journal in self.wasi.journals.clone() {
-                env.add_journal(journal);
+                builder.add_journal(journal);
             }
 
-            for snapshot_trigger in self.wasi.snapshot_on.iter().cloned() {
-                env.add_snapshot_trigger(snapshot_trigger);
+            for trigger in self.wasi.snapshot_on.iter().cloned() {
```

Comment:
small nit: this is a bit confusing, would be clearer with an `if wasi.snapshot_on.is_empty() { ... } else { ... }`

Reply:
Funny how most of the comments came from code I copy-pasted 😄